packages/common/src/helpers/calculateDynamicFontSize.ts

```diff
@@ -0,0 +1,80 @@
+import { PDFDocument, StandardFonts } from 'pdf-lib';
+import fontkit from '@pdf-lib/fontkit';
+import { TextSchemaWithData } from '../type';
+import { calculateTextWidthInMm } from './calculateTextWidthInMm';
+
+type DynamicFontSize = (
+  activeSchema: TextSchemaWithData,
+  fontData: {
+    [fontName: string]: {
+      data: string | ArrayBuffer | Uint8Array;
+    };
+  }
+) => Promise<number>;
+
+export const calculateDynamicFontSize: DynamicFontSize = async (activeSchema, fontData) => {
+  const DEFAULT_FONT_SIZE_IN_PIXELS = 18;
+  const DEFAULT_FONT_SIZE_SCALING_MIN = 100;
+  const DEFAULT_FONT_SIZE_SCALING_MAX = 100;
+  const DEFAULT_TOLERANCE = 3;
+  const DEFAULT_FONT_SIZE_ADJUSTMENT = 0.5;
+
+  const {
+    data,
+    fontName,
+    fontSize,
+    fontSizeScalingMax,
+    fontSizeScalingMin,
+    characterSpacing,
+    width,
+  } = activeSchema;
+
+  const baseFontSizeInPixels = fontSize ?? DEFAULT_FONT_SIZE_IN_PIXELS;
+  const minSizePercentage = fontSizeScalingMin ?? DEFAULT_FONT_SIZE_SCALING_MIN;
+  const maxSizePercentage = fontSizeScalingMax ?? DEFAULT_FONT_SIZE_SCALING_MAX;
+  const minFontSize = (minSizePercentage * baseFontSizeInPixels) / 100;
+  const maxFontSize = (maxSizePercentage * baseFontSizeInPixels) / 100;
+
+  const doc = await PDFDocument.create();
+  doc.registerFontkit(fontkit);
+
+  let font;
+
+  if (fontName === 'Helvetica') {
```

Comment:
Use [DEFAULT_FONT_NAME](https://github.com/pdfme/pdfme/blob/main/packages/common/src/helper.ts#L14).

Please move to constants.ts and export it.